apps/meteor/app/models/server/raw/Sessions.ts

```diff
@@ -761,17 +763,315 @@ export class SessionsRaw extends BaseRaw<ISession> {
+			},
+		] = await this.col.aggregate<{ docs: ISession[]; count: { total: number }[] }>(queryArray).toArray();
+
+		return { sessions, total, count, offset };
+	}
+
+	async getAllSessions(
+		search = '',
+		{ offset, count }: IPaginationOptions = { offset: 0, count: 10 },
+	): Promise<PaginatedResult<{ sessions: ISession & { _user: Pick<IUser, 'name' | 'username' | 'avatarETag' | 'avatarOrigin'> }[] }>> {
+		const searchQuery = search ? [{ $text: { $search: search } }] : [];
+
+		const matchOperator = {
+			$match: {
+				$and: [
+					...searchQuery,
+					{
+						loginToken: {
+							$exists: true,
+							$ne: '',
+						},
+					},
+					{
+						logoutAt: {
+							$exists: false,
+						},
+					},
+				],
+			},
+		};
+		const sortOperator = {
+			$sort: {
+				createdAt: -1,
+			},
+		};
+		const groupOperator = {
+			$group: {
+				_id: '$loginToken',
+				day: {
+					$first: '$day',
+				},
+				instanceId: {
+					$first: '$instanceId',
+				},
+				month: {
+					$first: '$month',
+				},
+				sessionId: {
+					$first: '$sessionId',
+				},
+				userId: {
+					$first: '$userId',
+				},
+				year: {
+					$first: '$year',
+				},
+				_updatedAt: {
+					$first: '$_updatedAt',
+				},
+				createdAt: {
+					$first: '$createdAt',
+				},
+				device: {
+					$first: '$device',
+				},
+				host: {
+					$first: '$host',
+				},
+				ip: {
+					$first: '$ip',
+				},
+				loginAt: {
+					$first: '$loginAt',
+				},
+				mostImportantRole: {
+					$first: '$mostImportantRole',
+				},
+				roles: {
+					$first: '$roles',
+				},
+			},
+		};
+		const limitOperator = { $limit: count };
+
+		const skipOperator = offset >= 1 ? [{ $skip: offset }] : [];
+
+		const lookupOperator = {
+			$lookup: {
+				from: 'users',
+				localField: 'userId',
+				foreignField: '_id',
+				as: '_user',
+			},
+		};
+		const unwindOperator = {
+			$unwind: {
+				path: '$_user',
+				preserveNullAndEmptyArrays: true,
+			},
+		};
+
+		const projectOperator = {
+			$project: {
+				_id: '$sessionId',
+				day: 1,
+				instanceId: 1,
+				month: 1,
+				sessionId: 1,
+				year: 1,
+				_updatedAt: 1,
+				createdAt: 1,
+				device: 1,
+				devices: 1,
+				host: 1,
+				ip: 1,
+				loginAt: 1,
+				mostImportantRole: 1,
+				roles: 1,
+				userId: 1,
+				_user: {
+					name: 1,
+					username: 1,
+					avatarETag: 1,
+					avatarOrigin: 1,
+				},
+			},
+		};
+
+		const facetOperator = {
+			$facet: {
+				docs: [sortOperator, ...skipOperator, limitOperator, lookupOperator, unwindOperator, projectOperator],
+				count: [
+					{
+						$count: 'total',
+					},
+				],
+			},
+		};
+
+		const queryArray = [matchOperator, sortOperator, groupOperator, facetOperator];
+
+		const [
+			{
+				docs: sessions,
+				count: [{ total } = { total: 0 }],
+			},
+		] = await this.col
+			.aggregate<{
+				docs: ISession & { _user: Pick<IUser, 'name' | 'username' | 'avatarETag' | 'avatarOrigin'> }[];
+				count: { total: number }[];
```

Comment:
This part could be its own type now, something like:
`WithItemCount<T> = T & { count: { total: number }[] }`

So that you can reuse it on other places

Reply:
I agree. And since not all session attributes are being projected, the resulting type is not a full `ISession`, in this case it's better to set the type picking only the attributes that are actually projected.